### ISSUE TITLE
Proposal: facilitate getting image sizes added to markdown

### ISSUE THREAD

**Issue by ell1e** [go-gitea/gitea#27339]
### Feature Description

GitLab has image sizes now, has had for a while: [(Click for Gitlab image size spec)](https://docs.gitlab.com/ee/user/markdown.html#images)

    ![alt text](img/markdown_logo.png){width=100 height=100px}

The huge problem of no image sizes is, that it makes any High-DPI screenshots in technical documentation use impossible. Any logos of a modern project will also likely be High-DPI, so this could affect pretty much every future README that is slightly more polished.

**Why bother with that in markdown, in detail:** The source of the trouble is that browsers ignore DPI attributes, so any included High-DPI image is going to be way too large unless a proper, smaller display size can be specified. This is not just a style and polish concern, since a High-DPI image can easily be 2x or even 3x too large and be so giant, that it won't even fit on any user's screen and makes the documentation just plain unreadable. So this is not a minor styling concern that would usually have no place in markdown, given markdown is otherwise style agnostic.

Also, this affects basically all documentation of all projects with mobile screenshots or logos, which should be possibly a huge amount or even majority of projects, and most importantly also not just giant corporation ones that might want to be on Gitlab or Github or similar anyway but also small self-hosted ones.

**Why make a ticket here and not upstream?** Simply because it seems like [goldmark seems unlikely to add this into the core project](https://github.com/yuin/goldmark/discussions/331), so if this is ever to arrive in gitea probably someone would need to add an extension or a different markdown library would need to be used. Also, [the old Gitea ticket was closed](https://github.com/go-gitea/gitea/issues/21473) in favor of the upstream discussion but that discussion doesn't seem to be going anywhere.

I hope this can eventually arrive in gitea.

**Comment by silverwind** [comment]
This is a good idea. I'm also annoyed that uploaded images are rendered too large on GitHub and Gitea because they don't respect DPI scaling.

We should check [window.devicePixelRatio](https://developer.mozilla.org/en-US/docs/Web/API/Window/devicePixelRatio) when uploading images in JS and reduce the image size with these attributes.

The drawback of this method is that we assume the device uploading the image is also the one that took the image, which is not always correct, but I guess it's better then not doing anything.

**Comment by ell1e** [comment]
As far as I know, the metadata for images usually doesn't contain the correct pixel ratio. So what you propose isn't possible.

(As for the UI as a whole, it's already scaled correctly by the browser, including applying `window.devicePixelRatio`. The pixel measurements given above are virtual pixels at a 1.0 DPI scale factor that get adjusted by the browser. So that part is working fine out of the box and doesn't need adjustments either.)

**Comment by silverwind** [comment]
My thought was that most content uploaded is screenshots and if someone with for example pixel ratio 2 uploads a image at 1000x1000, we would reduce that to 500x500 for rendering, so the image would be rendered with the correct amount of CSS pixels.

**Comment by ell1e** [comment]
Right, but you don't know each screenshot's pixel ratio. That's what the width/height annotations are for. (The pixel ratio of the viewer's screen is already taken into account by every modern browser so that's a non-issue.)

**Comment by ell1e** [comment]
My apologies for the confusion on my side, I get what you mean now. It sounds useful as an addition.

**Comment by ell1e** [comment]
Maybe there could just be a scale adjuster in the "Preview" tab of the comments? Then if someone checks the preview and notices it's way too giant or too small, they could still fix it. (In addition to detecting it via JavaScript as you say.)

In any case, I really hope the Markdown syntax extension for image sizes can happen for README's and such. The problem is also if I'm using pandoc or other tools to also generate PDFs and other formats from markdown docs in a project, I can't use image sizes purely to not break the display on Gitea with spurious weird fragments on Gitea. So anyone who has their project and docs on Gitea will kind of be held back of also fixing this for other channels.

**Comment by kalleswe** [comment]
Why isn't this added yet? We need to use it today.

**Comment by eeyrjmr** [comment]
> Why isn't this added yet? We need to use it today.

Patches welcome
